packages/sanity/src/core/form/inputs/files/common/PlaceholderText.tsx

```diff
@@ -34,27 +37,27 @@ export function PlaceholderText(props: Props) {
   }, [directUploads, hoveringFiles, isFileType, readOnly, rejectedFilesCount])
 
   const messageText = useMemo(() => {
-    let message = `Drag or paste ${type} here`
-
     if (!directUploads) {
-      return `Can't upload files here`
+      return t('inputs.files.common.placeholder.upload-not-supported')
     }
 
     if (readOnly) {
-      message = 'Read only'
+      return t('inputs.files.common.placeholder.read-only')
     }
 
     if (hoveringFiles && directUploads && !readOnly) {
       if (acceptedFiles.length > 0) {
-        message = `Drop to upload ${type}`
+        return t('inputs.files.common.placeholder.drop-to-upload', {type})
```

Comment:
I think this should use the [`context` option](https://www.i18next.com/translation-function/context) in order to allow translation of the type. Right now, my hunch is that this will show _"Slipp for å laste opp image"_ in Norwegian, for instance. By passing the type as context, you can define `inputs.files.common.placeholder.drop-to-upload_image` and `inputs.files.common.placeholder.drop-to-upload_file` separately

```suggestion
        return t('inputs.files.common.placeholder.drop-to-upload', {context: type})
```

Reply:
Ooh, that's a great callout, wasn't aware of the context option. Thanks!